packages/seacas/libraries/ioss/src/Ioss_Decomposition.C

```diff
@@ -158,106 +158,110 @@ namespace Ioss {
     return valid_methods;
   }
 
-  size_t get_all_block_ioss_element_size(const std::vector<BlockDecompositionData> &blocks)
+  size_t ElementBlockBatchOffset::get_ioss_element_size(const std::vector<int64_t>& blockSubsetIndex) const
   {
     size_t count = 0;
 
-    for (const Ioss::BlockDecompositionData &block : blocks) {
-      // Determine total number of ioss decomp elements
+    // Determine total number of ioss subset decomp elements
+    for(int64_t i : blockSubsetIndex) {
+      const Ioss::BlockDecompositionData& block = m_data[i];
       count += (block.importMap.size() + block.localMap.size());
     }
 
     return count;
   }
 
-  size_t get_all_block_ioss_offset_size(const std::vector<BlockDecompositionData> &blocks,
-                                        const std::vector<int> &block_field_component_count)
+  size_t ElementBlockBatchOffset::get_ioss_offset_size(const std::vector<int64_t>& blockSubsetIndex,
+                                                       const std::vector<int>& blockSubsetFieldComponentCount) const
   {
     size_t count = 0;
 
-    for (size_t blk_seq = 0; blk_seq < blocks.size(); blk_seq++) {
-      const Ioss::BlockDecompositionData &block = blocks[blk_seq];
-      // Determine total number of ioss decomp entries based on field component count per block.
-      count +=
-          block_field_component_count[blk_seq] * (block.importMap.size() + block.localMap.size());
+    for(size_t i = 0; i < blockSubsetIndex.size(); i++) {
```

Comment:
In the loop starting at 166, you loop over `blockSubsetIndex`, but here you do a loop on the size of blockSubsetIndex, get the `blk_seq` from that and then the block from `m_data`. Is there a reason for the extra indirection here?

Reply:
Yes, this is necessary because the blockSubsetFieldComponentCount vector needs to be referenced by an index `i` but that was not necessary in the for-loop on line 166